native/c/zlogger_metal.c

```diff
@@ -0,0 +1,312 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+#include "dcbd.h"
+#include "zlogger_metal.h"
+#include "zam.h"
+#include "zutm.h"
+#include "zwto.h"
+
+static struct
+{
+  char log_path[256]; /* Log file path */
+  int min_level;      /* Minimum log level */
+  int initialized;    /* Initialization flag */
+  int use_wto;        /* Whether to use WTO for logging */
+  int use_dd;         /* Whether DD is available for file I/O */
+  IO_CTRL *log_io;    /* BSAM I/O control block for log DD */
+} g_metal_logger = {0};
+
+static const char *g_level_strings[] = {
+    "TRACE", "DEBUG", "INFO", "WARN", "ERROR", "FATAL", "OFF"};
+
+/**
+ * Get current timestamp for Metal C using z/OS STCK instruction
+ */
+static void ZLGTIME(char *buffer, int buffer_size)
+{
+#if defined(__IBM_METAL__)
+  /* Use z/OS STCK (Store Clock) instruction for high-precision timestamp */
+  unsigned long long clock_value = 0;
+
+  __asm(" STCK %0" : "=m"(clock_value) : : "cc");
+
+  /* Convert STCK value to a simple readable format */
+  /* STCK returns microseconds since 1900-01-01 00:00:00 GMT */
+  /* For simplicity, we'll just show the lower bits as a hex timestamp */
+  sprintf(buffer, "%016llX", clock_value);
+#else
+  /* Fallback for non-Metal C compilation */
+  strncpy(buffer, "TIMESTAMP", buffer_size - 1);
+  buffer[buffer_size - 1] = '\0';
+#endif
+}
+
+/**
+ * Write message using WTO (Write To Operator)
+ */
+static int ZLGWRWTO(int level, const char *message)
+{
+  /* Ensure WTO_BUF is aligned on fullword boundary for Metal C */
+  WTO_BUF wto_buf __attribute__((aligned(4))) = {0};
```

Comment:
if you can, `ATTRIBUTE` from `ztype` instead of `__attribute__` for off platform IDEs and avoiding false errors

Reply:
Addressed in 99a58db6